### PR TITLE
Docker relevant upgrade

### PR DESCRIPTION
**Pull request by vttranlina**

### CODE REVIEW

**Review by chibenwa — COMMENTED** [review]
Please add a section to the unreleased changelog.

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,8 +4,11 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 
 ## [Unreleased]
-
-No changes yet.
+### Changed
```
Comment (chibenwa) [inline CHANGELOG.md:7]:
```suggestion
### Third party

We recommend upgrading the following third party dependencies:
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,8 +4,11 @@ All notable changes to this project will be documented in this file.
...
+- Upgrade docker relevant: rabbitmq -> 3.9.18
+- Upgrade docker relevant: tika -> 1.28.2 
+- Upgrade docker relevant: spamassassin -> 3.4.6-1 T
+- Upgrade docker relevant: nginx -> 1.22
```
Comment (chibenwa) [inline CHANGELOG.md:11]:
```suggestion
```

Rationals: that's a testing dependency

**Review by chibenwa — APPROVED** [review]

**Comment by Arsnael** [comment]
```
org.apache.james.backends.es.v7.ClientProviderImplConnectionAuthESOverrideTrustStoreTest.org.apache.james.backends.es.v7.ClientProviderImplConnectionAuthESOverrideTrustStoreTest
Stacktrace

java.lang.ExceptionInInitializerError
	at org.apache.james.backends.es.v7.ClientProviderImplConnectionAuthESOverrideTrustStoreTest.<clinit>(ClientProviderImplConnectionAuthESOverrideTrustStoreTest.java:34)
Caused by: org.testcontainers.containers.ContainerLaunchException: Container startup failed
	at org.apache.james.backends.es.v7.ClientProviderImplConnectionAuthESOverrideTrustStoreTest.<clinit>(ClientProviderImplConnectionAuthESOverrideTrustStoreTest.java:34)
Caused by: org.testcontainers.containers.ContainerFetchException: Can't get Docker image: RemoteDockerImage(imageName=<resolving>, imagePullPolicy=DefaultPullPolicy())
	at org.apache.james.backends.es.v7.ClientProviderImplConnectionAuthESOverrideTrustStoreTest.<clinit>(ClientProviderImplConnectionAuthESOverrideTrustStoreTest.java:34)
Caused by: com.github.dockerjava.api.exception.DockerClientException: Could not build image: COPY failed: file not found in build context or excluded by .dockerignore: stat nginx-conf/: file does not exist
	at org.apache.james.backends.es.v7.ClientProviderImplConnectionAuthESOverrideTrustStoreTest.<clinit>(ClientProviderImplConnectionAuthESOverrideTrustStoreTest.java:34)

Standard Output

09:13:13.048 [ERROR] 🐳 [localhost/testcontainers/z7qhyhayk2wkxc81] - COPY failed: file not found in build context or excluded by .dockerignore: stat nginx-conf/: file does not exist
```
Related to nginx upgrade or not?

```
org.apache.james.backends.rabbitmq.ReactorRabbitMQChannelPoolTest.notUsedChannelShouldBeClosedWhenPoolIsClosed
Error Message

Expecting value to be false but was true

Stacktrace

org.opentest4j.AssertionFailedError: 

Expecting value to be false but was true
	at org.apache.james.backends.rabbitmq.ReactorRabbitMQChannelPoolTest.notUsedChannelShouldBeClosedWhenPoolIsClosed(ReactorRabbitMQChannelPoolTest.java:117)
```
Related to rabbitmq upgrade or not?

**Comment by Arsnael** [comment]
I saw the same rabbitmq error on an other build so it's more instability, not related to this PR

**Comment by vttranlina** [comment]
> I saw the same rabbitmq error on an other build so it's more instability, not related to this PR

yes, I see the latest master has some diff related to Rabbit, I should rebase this pr

**Review by Arsnael — APPROVED** [review]

**Comment by chibenwa** [comment]
https://ci-builds.apache.org/job/james/job/ApacheJames/job/PR-1027/8/testReport/org.apache.james.transport.mailets/SpamAssassinIntegrationTest/org_apache_james_transport_mailets_SpamAssassinIntegrationTest/

```
org.apache.james.transport.mailets.SpamAssassinIntegrationTest.org.apache.james.transport.mailets.SpamAssassinIntegrationTest

Stacktrace

org.testcontainers.containers.ContainerLaunchException: Container startup failed
Caused by: org.testcontainers.containers.ContainerFetchException: Can't get Docker image: RemoteDockerImage(imageName=dinkel/spamassassin:3.4.6-1, imagePullPolicy=DefaultPullPolicy())
Caused by: com.github.dockerjava.api.exception.NotFoundException: 
Status 404: {"message":"manifest for dinkel/spamassassin:3.4.6-1 not found: manifest unknown: manifest unknown"}
```

Looks related...

File: server/testing/src/main/java/org/apache/james/util/docker/Images.java [hunk 3]
```diff
@@ -21,12 +21,12 @@
...
+    String RABBITMQ = "rabbitmq:3.9.18-management";
     String ELASTICSEARCH_2 = "elasticsearch:2.4.6";
     String ELASTICSEARCH_6 = "docker.elastic.co/elasticsearch/elasticsearch:6.3.2";
     String ELASTICSEARCH_7 = "docker.elastic.co/elasticsearch/elasticsearch:7.10.2";
-    String NGINX = "nginx:1.15.1";
-    String TIKA = "apache/tika:1.26";
-    String SPAMASSASSIN = "dinkel/spamassassin:3.4.0";
+    String NGINX = "nginx:1.22";
+    String TIKA = "apache/tika:1.28.2";
+    String SPAMASSASSIN = "dinkel/spamassassin:3.4.6-1";
```
Comment (quantranhong1999) [inline server/testing/src/main/java/org/apache/james/util/docker/Images.java:30]:
```suggestion
    String SPAMASSASSIN = "instantlinux/spamassassin:3.4.6-1";
```

**Comment by vttranlina** [comment]
rebase latest master

**Pull request merged**